code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -359,24 +361,60 @@
 		P.fire()
 		return P
 
-/mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
-	return iswallturf(T) || ismineralturf(T)
 
-/mob/living/simple_animal/hostile/proc/DestroySurroundings()
+/mob/living/simple_animal/hostile/proc/DestroyStructuresCycle(turf/T)
+	var/dir_to_target = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
+	var/dir_from_target = get_dir(targets_from, get_cardinal_step_away(targets_from, target))
+
+	var/list/dir_list = list(dir_from_target, get_next_cardinal_dir(dir_from_target, 1), get_next_cardinal_dir(dir_to_target, 1), dir_to_target)
+
+	for(dir_list.len, dir_list.len > 0, dir_list.len--)
```

Comment:
are the faster loop formats in contributor.md?
in any case, this loop is very odd and you can probably do it faster and find the dirs you need to attack much more easily.

Reply:
i cant find contributor.md
i also dont know much about optimization, but if there are predefined loop formats, id be happy to try to apply them

Reply:
nvm found it, its contributing.md